src/EPG.cpp

```diff
@@ -31,7 +31,9 @@ EPG::EPG(const std::shared_ptr<InstanceSettings>& settings, Request& request, Re
 PVR_ERROR EPG::GetEPGForChannel(int channelUid, time_t start, time_t end, kodi::addon::PVREPGTagsResultSet& results)
 {
   std::pair<bool, bool> channelDetail;
+  std::lock_guard<std::recursive_mutex> lock(m_channels.m_channelMutex);
   channelDetail = m_channels.m_channelDetails[channelUid];
+  std::lock_guard<std::recursive_mutex> unlock(m_channels.m_channelMutex);
```

Comment:
Just put the protected line in a block instead. You should never need to unlock:

```
  {
    std::lock_guard<std::recursive_mutex> lock(m_channels.m_channelMutex);
    channelDetail = m_channels.m_channelDetails[channelUid];
  }
```

Reply:
Thanks I didn't know that was prefered